Use callback wrapper for LocaleChoiceType choices

wimulkeman opened the pull request:
Using the callback wrapper to resolve the LocaleChoiceType keeping old entity objects during tests where both the form is build and submit through the Symfony PhpUnit WebTestCase.

Also see https://github.com/Sylius/Sylius/issues/16572

| Q               | A
|-----------------|-----
| Branch?         | 1.13
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes #https://github.com/Sylius/Sylius/issues/16572
| License         | MIT

wimulkeman committed:
fix(locales): use callback wrapper for choices

Using the callback wrapper to resolve the LocaleChoiceType keeping old entity objects during tests where both the form is build and submit through the Symfony PhpUnit WebTestCase.

Also see https://github.com/Sylius/Sylius/issues/16572

Fixes #16572

wimulkeman commented:
One of the build steps failed, but it seems to be a timeout rather than a code style or test failure. I could not find an option to rerun the specific step.

mpysiak commented:
@wimulkeman I've triggered a rerun on failed steps

Rafikooo commented on src/Sylius/Bundle/LocaleBundle/Form/Type/LocaleChoiceType.php:
```diff
@@ -40,7 +41,7 @@ public function configureOptions(OptionsResolver $resolver): void
 
         $resolver->setDefaults([
-            'choices' => $this->localeRepository->findAll(),
+            'choices' => fn (Options $options): array => $this->localeRepository->findAll(),
```
```suggestion
            'choices' => fn (): array => $this->localeRepository->findAll(),
```
It appears that the `$options` argument is not being utilized. Could you clarify the reason for including it here?

wimulkeman replied:
The main reason I opted to include the `$options` is to mimic similar setups used in other classes:

CurrencyChoiceType:
https://github.com/Sylius/Sylius/blob/1.12/src/Sylius/Bundle/CurrencyBundle/Form/Type/CurrencyChoiceType.php#L40

ChannelChoiceType:
https://github.com/Sylius/Sylius/blob/1.12/src/Sylius/Bundle/ChannelBundle/Form/Type/ChannelChoiceType.php#L40

But I would not mind at all to drop it and implement your suggestion if preferred.

Wojdylak replied:
If it is not a problem, please make it without the `options` parameter.

mpysiak replied:
I think `options` parameter is necessary even if it is not used

wimulkeman committed:
Remove unused argument

Removing the unused $options argument for the variable is not used in the function.

Co-authored-by: Rafał Jaskulski <40125720+Rafikooo@users.noreply.github.com>

wimulkeman commented:
The build failed with the error `InvalidArgumentException: Time has run out and the assertion has not passed yet. in src/Sylius/Behat/Service/Helper/JavaScriptTestHelper.php:89` for multiple tests. I suspect the build hung on something other then the recent code change.

wimulkeman committed:
Revert "Remove unused argument
"

This reverts commit d97db7ca28c93348677ca24ee482e5a0c7a544cd.

Wojdylak commented:
Thank you, @wimulkeman!
